ddl/generated_column.go

```diff
@@ -50,10 +50,29 @@ func verifyColumnGeneration(colName2Generation map[string]columnGenerationInDDL,
 	return nil
 }
 
+// verifyColumnGenerationSingle is for ADD GENERATED COLUMN, we just need verify one column itself.
+func verifyColumnGenerationSingle(dependColNames map[string]struct{}, cols []*table.Column, position *ast.ColumnPosition) error {
+	// cause the added column hasn't existed, should derive it's offset from ColumnPosition
+	pos, err := findPositionRelativeColumn(cols, position)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	// mysql then will check relative generated column should be prior to it, so it should be in one loop
+	for _, col := range cols {
+		if _, ok := dependColNames[col.Name.L]; ok {
+			if col.IsGenerated() && col.Offset >= pos {
+				// Generated column can refer only to generated columns defined prior to it.
+				return errGeneratedColumnNonPrior.GenWithStackByArgs()
+			}
+		}
+	}
+	return nil
+}
+
 // checkDependedColExist ensure all depended columns exist.
-//
 // NOTE: this will MODIFY parameter `dependCols`.
 func checkDependedColExist(dependCols map[string]struct{}, cols []*table.Column) error {
+	// mysql will first check all depended column exist
```

Comment:
```suggestion
	// Checks all depended columns exist
```
We don't need to comment 'MySQL do what' unless the logic is confusing/counter-intuitive